File: apps/api/app/schemas/billing.py
from datetime import datetime
from typing import Optional, List, Dict, Any
from uuid import UUID
from pydantic import BaseModel


class BillableItemResponse(BaseModel):
    id: UUID
    visit_id: UUID
    code: str
    category: str
    description: Optional[str] = None
    start_ms: int
    end_ms: int
    minutes: int
    evidence: List[Dict[str, Any]]
    is_approved: bool
    is_flagged: bool
    flag_reason: Optional[str] = None
    adjusted_minutes: Optional[int] = None
    adjustment_reason: Optional[str] = None
    created_at: datetime

    class Config:
        from_attributes = True


class BillableItemUpdate(BaseModel):
    is_approved: Optional[bool] = None
    is_flagged: Optional[bool] = None
    flag_reason: Optional[str] = None
    adjusted_minutes: Optional[int] = None
    adjustment_reason: Optional[str] = None


class BillingResponse(BaseModel):
    visit_id: UUID
    items: List[BillableItemResponse]
    total_minutes: int
    total_adjusted_minutes: int
    categories: Dict[str, int]  # category -> minutes

File: ios-app/PalmCareAI/Models/Billing.swift
import Foundation

struct UploadResponse: Codable {
    let id: String
    let visit_id: String
    let s3_key: String
    let original_filename: String?
    let content_type: String?
    let file_size_bytes: Int?
    let duration_ms: Int?
    let status: String
    let created_at: String
}

struct VisitBillablesResponse: Codable {
    let items: [BillableItem]?
    let total_minutes: Double?
    let total_adjusted_minutes: Double?
    let categories: [String: AnyCodable]?
}

struct BillableItem: Codable, Identifiable {
    let id: String
    let visit_id: String?
    let code: String?
    let category: String?
    let description: String?
    let start_ms: Int?
    let end_ms: Int?
    let minutes: Double?
    let evidence: AnyCodable?
    var is_approved: Bool?
    var is_flagged: Bool?
    let flag_reason: String?
    let adjusted_minutes: Double?

    /// Assessment / intake recommendation (not timed visit work).
    var isRecommendation: Bool {
        let reason = (flag_reason ?? "").lowercased()
        return reason.contains("recommended from assessment")
            || reason.contains("not timed visit")
    }

    var timeLabel: String {
        if isRecommendation {
            return "Recommended"
        }
        guard let minutes, minutes > 0 else {
            return "No time logged"
        }
        if minutes == 1 {
            return "1 min"
        }
        return "\(Int(minutes)) min"
    }
}
